message_ix/__init__.py

```diff
@@ -3,8 +3,11 @@
 
 import ixmp.model_settings as model_settings
 
-from message_ix import default_paths
-from message_ix.core import *
+from . import default_paths
+from .core import (  # noqa: F401
```

Comment:
This seems like an API change (based on how users can import things) - do you think it is sufficiently important to document? Was there a reason to make this change?

Reply:
If I do something like this:
```py
>>> import message_ix
>>> dir(message_ix)
>>> len(dir(message_ix))
```

…then there are 39 items before the change, and 28 afterwards. The excluded items are: `collections, copy, isscalar, itertools, ixmp, logger, np, pd, pd_read, pd_write, warnings`.

Some of these are reimports. We should encourage the user to do `import numpy as np; np.ones(...)` and never `import message_ix; message_ix.np.ones(...)` and so on.

The only two defined by message_ix itself are `message_ix.utils.isscalar` and `message_ix.utils.logger`. Do we mean to promise that those are available as `message_ix.isscalar` and `message_ix.logger`? If so, I can add tests to that effect, and they can be imported directly from `message_ix.utils`. IMHO it's clearer to do this explicitly than implicitly/indirectly with `from message_ix.core import *`.

Reply:
I agree in general - thus would not consider this blocking for the PR. AFAIK, this is common practice in python libraries (i.e., importing submodules in their intirety in `__init__.py`). This is abused in the usage of some libraries though (looking at `from pyomo.environ import *`), but within the namespace I think its ok.

In any case, no problem with the change =)